File: backend/user_details/enums/sleep_quality.py
from enum import Enum


class SleepQuality(str, Enum):
    POOR = "poor"
    FAIR = "fair"
    GOOD = "good"
    EXCELLENT = "excellent"

    def get_sleep_quality_factor(self):
        return {
            SleepQuality.POOR: -0.03,
            SleepQuality.FAIR: 0.0,
            SleepQuality.GOOD: 0.02,
            SleepQuality.EXCELLENT: 0.03,
        }[self]

File: frontend/lib/models/user_details/sleep_quality.dart
enum SleepQuality {
  poor(0, 'poor'),
  fair(1, 'fair'),
  good(2, 'good'),
  excellent(3, 'excellent');

  final int value;
  final String nameStr;

  const SleepQuality(this.value, this.nameStr);

  String toJson() => nameStr;

  static SleepQuality fromJson(String value) {
    return SleepQuality.values.firstWhere(
      (e) => e.nameStr == value.toUpperCase(),
      orElse: () => throw ArgumentError('Unknown sleep quality: $value'),
    );
  }

  int toInt() => value;

  static SleepQuality fromInt(int value) {
    return SleepQuality.values.firstWhere(
      (e) => e.value == value,
      orElse: () => throw ArgumentError('Invalid sleep quality value: $value'),
    );
  }
}
